plugins/animate/fire/fire.cpp

```diff
@@ -88,7 +90,46 @@ class FireTransformer : public wf::view_transformer_t
         p.life = 1;
         p.fade = random(0.1, 0.6);
 
-        p.color = {random(0.4, 1), random(0.08, 0.2), random(0.008, 0.018), 1};
+        wf::color_t color_setting = fire_color;
+        bool random_colors = random_fire_color;
+        
+        float r;
+        float g;
+        float b;
+
+        if (!random_colors) {
+            // The calculation here makes the variation lower at darker values
+            float randomize_amount_r = (color_setting.r * 0.857) / 2;
+            float randomize_amount_g = (color_setting.g * 0.857) / 2;
+            float randomize_amount_b = (color_setting.b * 0.857) / 2;
+
+            r = random(color_setting.r - randomize_amount_r, 
+                       std::min(color_setting.r + randomize_amount_r, 
+                       1.0));
+            g = random(color_setting.g - randomize_amount_g, 
+                       std::min(color_setting.g + randomize_amount_g, 
+                       1.0));
+            b = random(color_setting.b - randomize_amount_b, 
+                       std::min(color_setting.b + randomize_amount_b, 
+                       1.0));
+        } else {
+            r = random(0, 1);
+            g = random(0, 1);
+            b = random(0, 1);
+
+            // Squrae the values several times to make them less of a blinding light and more of a spray of colors
+            for (int i = 0; i < 4; i++) {
```

Comment:
this seems more like `r = 2 * pow(r, 16)` which is much shorter and clearer imo.

Reply:
Fixed it now